backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopHighLevelCompiler.scala

```diff
@@ -140,12 +139,7 @@ final class BloopHighLevelCompiler(
             case NonFatal(t) =>
               // If scala compilation happens, complete the java promise so that it doesn't block
               JavaCompleted.tryFailure(t)
-
-              t match {
-                case _: NullPointerException if cancelPromise.isCompleted =>
-                  throw new InterfaceCompileCancelled(Array(), "Caught NPE when compilation was cancelled!")
```

Comment:
This seems to be useful to just wrap the exception, which doesn't seem neccessary.